lib/backend/k8s/resources/customresource.go

```diff
@@ -155,19 +181,23 @@ func (c *customK8sResourceClient) Apply(kvp *model.KVPair) (*model.KVPair, error
 	})
 	logContext.Debug("Apply custom Kubernetes resource")
 
-	// Attempt and Update and roll back to a Create if the resource does
-	// not exist.  We only log debug here since the Update and Create will
-	// also log.
-	updated, err := c.Update(kvp)
+	// Attempt to Create and do an Update if the resource already exists.
+	// We only log debug here since the Create and Update will also log.
+	// Can't set Revision while creating a resource.
+	updated, err := c.Create(&model.KVPair{
```

Comment:
Create fails if revision is provided. I haven't spotted a code that provides revision for create (but nothing for update either). 
 > Perhaps we want to skip "Create" if the revision is set?

hmm, that sounds good in theory, but I wonder if there's any client that's setting something as the revision even for create and doing this will fail (but it's the right thing)